Patch-Source: https://github.com/sharkdp/fd/pull/1867
--
From 5db9cba3ebedff386f289c7beaa05ab025880be6 Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Thu, 1 Jan 2026 08:52:06 -0500
Subject: [PATCH] feat(build): enable jemalloc on musl and riscv64

Alpine Linux vendors a custom patch to enable jemalloc for all arches, which seems to work fine in practice.

Signed-off-by: Aster Boese <asterboese@mailbox.org>
---
 Cargo.toml  | 2 +-
 src/main.rs | 2 --
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 192cdb10e..8fff897a3 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,7 +67,7 @@ libc = "0.2"
 # Catalina. See https://github.com/sharkdp/fd/issues/498 for details.
 # This has to be kept in sync with src/main.rs where the allocator for
 # the program is set.
-[target.'cfg(all(not(windows), not(target_os = "android"), not(target_os = "macos"), not(target_os = "freebsd"), not(target_os = "openbsd"), not(target_os = "illumos"), not(all(target_env = "musl", target_pointer_width = "32")), not(target_arch = "riscv64")))'.dependencies]
+[target.'cfg(all(not(windows), not(target_os = "android"), not(target_os = "macos"), not(target_os = "freebsd"), not(target_os = "openbsd"), not(target_os = "illumos")))'.dependencies]
 tikv-jemallocator = {version = "0.6.0", optional = true}
 
 [dev-dependencies]
diff --git a/src/main.rs b/src/main.rs
index fafb3b900..fd158e6e7 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -45,8 +45,6 @@ use crate::regex_helper::{pattern_has_uppercase_char, pattern_matches_strings_wi
     not(target_os = "freebsd"),
     not(target_os = "openbsd"),
     not(target_os = "illumos"),
-    not(all(target_env = "musl", target_pointer_width = "32")),
-    not(target_arch = "riscv64"),
     feature = "use-jemalloc"
 ))]
 #[global_allocator]
